core/domain/user_domain_test.py

```
@@ -339,6 +339,20 @@ def test_validate_non_str_display_alias_raises_error(self) -> None:
         ):
             self.user_settings.validate()
 
+    # TODO(#13059): Here we use MyPy ignore because after we fully type the
+    # codebase we plan to get rid of the tests that intentionally test wrong
+    # inputs that we can normally catch by typing.
+    def test_validate_non_str_profile_name_for_certificate_raises_error(
+        self,
+    ) -> None:
+        self.user_settings.profile_name_for_certificate = 0  # type: ignore[assignment]
+        with self.assertRaisesRegex(
```

Ruff (0.15.21)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)